oshmem/mca/sshmem/ucx/sshmem_ucx_shadow.c

```diff
@@ -107,6 +107,66 @@ static void sshmem_ucx_shadow_merge_blocks(sshmem_ucx_shadow_allocator_t *alloca
     }
 }
 
+
+
+int sshmem_ucx_shadow_realloc(sshmem_ucx_shadow_allocator_t *allocator,
+                              unsigned count, unsigned old_index, unsigned *index,
+                              int *inplace)
+{
+    sshmem_ucx_shadow_alloc_elem_t *end  = &allocator->elems[allocator->num_elems];
+    sshmem_ucx_shadow_alloc_elem_t *elem = &allocator->elems[old_index];
+    sshmem_ucx_shadow_alloc_elem_t *next = &elem[elem->block_size];
+    unsigned old_count                   = elem->block_size;
+
+    assert(count > 0);
+    assert(!sshmem_ucx_shadow_is_free(elem));
+
+    *inplace = 1;
+
+    if (count == old_count) {
+        *index = old_index;
+        return OSHMEM_SUCCESS;
+    }
+
+    if (count < old_count) {
+        /* requested block is shorter than allocated block
+         * then just cut current buffer */
+        sshmem_ucx_shadow_set_elem(elem + count,
+                                   SSHMEM_UCX_SHADOW_ELEM_FLAG_FREE,
+                                   elem->block_size - count);
+        elem->block_size = count;
+        *index           = old_index;
+        sshmem_ucx_shadow_merge_blocks(allocator);
+        return OSHMEM_SUCCESS;
+    }
+
+    assert(count > old_count);
+
+    /* try to check if next element is free & has enough length */
+    if ((next < end)                    && /* non-last element? */
```

Comment:
can we initialize 'next' and 'end' here? since we don't always use them
also no need for many spaces before '&&'

Reply:
ok, fixed